### PR TITLE
[SPARK-33695][BUILD] Upgrade to jackson to 2.10.5 and jackson-databind to 2.10.5.1

### PR DESCRIPTION
**Pull request by n-marion**
### What changes were proposed in this pull request?

Upgrade the jackson dependencies to 2.10.5 and jackson-databind to 2.10.5.1

### Why are the changes needed?

Jackson dependency has vulnerability CVE-2020-25649.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Existing unit tests.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -169,7 +169,8 @@
     <!-- for now, not running scalafmt as part of default verify pipeline -->
     <scalafmt.skip>true</scalafmt.skip>
     <codehaus.jackson.version>1.9.13</codehaus.jackson.version>
-    <fasterxml.jackson.version>2.10.0</fasterxml.jackson.version>
+    <fasterxml.jackson.version>2.10.5</fasterxml.jackson.version>
+    <fasterxml.jackson-databind.version>2.10.5.1</fasterxml.jackson-databind.version>
```
Comment (maropu) [inline pom.xml:173]: Why you added a new entry here?

File: pom.xml [hunk 2]
```diff
@@ -169,7 +169,8 @@
     <!-- for now, not running scalafmt as part of default verify pipeline -->
     <scalafmt.skip>true</scalafmt.skip>
     <codehaus.jackson.version>1.9.13</codehaus.jackson.version>
-    <fasterxml.jackson.version>2.10.0</fasterxml.jackson.version>
+    <fasterxml.jackson.version>2.10.5</fasterxml.jackson.version>
```
Comment (maropu) [inline pom.xml:172]: We cannot bump up it to 2.12.0?

**Comment by maropu** [comment]
ok to test

**Comment by maropu** [comment]
Also, you need to update the manifest files:
```
./dev/test-dependencies.sh --replace-manifest
```

File: pom.xml (hunk 1 above)
Comment (n-marion) [reply pom.xml:173]: Jackson does this often and databind sometimes has a different subrevision. This follows the previous style before the recent update.

File: pom.xml (hunk 2 above)
Comment (n-marion) [reply pom.xml:172]: Looking over Jackson's release information both [Jackson Release 2.11](https://github.com/FasterXML/jackson/wiki/Jackson-Release-2.11) and [Jackson Release 2.12](https://github.com/FasterXML/jackson/wiki/Jackson-Release-2.12) showed a large amounts of changes; and although I could attempt a build to make sure compilation fails, I'd personally be uncertain that unit tests would be sufficient enough to ensure compatibility.

**Review by srowen — APPROVED** [review]

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thanks, @n-marion , @maropu , @srowen .
Merged to master for Apache Spark 3.2.0.

**Pull request closed without merging**